### PR TITLE
Code Review - MoneyCompass - Rails Capstone Project

### PR DESCRIPTION
**Pull request by luigirazum**
Create a Ruby on Rails application that allows the user to:
- register and log in, so that the data is private to them.
- introduce new transactions associated with a category.
- see the money spent on each category.

### General requirements

- [X] Make sure that there are no linter errors.
- [X] Make sure that you used correct gitflow.
- [X] Make sure that you documented your work in a professional way.

#### Pages Created
- [X] Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- [X] Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- [X] Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- [X] Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- [X] "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- [X] "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- [X] Create unit and integration tests for all the most important components of your RoR application.

#### Technical requirements
- [X] You should use Postgres as your database.
- [X] You should use devise for authentication.
- [X] You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- [X] The project should be deployed and accessible online.

#### Documentation
- [X] Video [presentation](https://screenrec.com/share/HNntTVAqLQ) of the project.

### CODE REVIEW

File: README.md
Comment (shahadat3669) [inline README.md:1]: 👍 Excellent progress on your project! However, there are a couple of essential steps to complete for a comprehensive submission:

- [x] **Project Presentation Video**: As per project requirements, kindly record a short video where you present your project. Share insights into the functionalities, key features, and any interesting aspects you'd like to highlight. This video is an excellent opportunity to showcase your work and provide context to the reviewers.

- [x] **Deployment Link in Readme**: Ensure you deploy your project on Render or another server, and include the deployment link in the readme file. This allows reviewers to access your project easily. Follow the documentation to understand the deployment process thoroughly.

Thank you for your efforts, and I look forward to reviewing your completed project submission! 🚀🎥

File: spec/features/splash_screen_spec.rb
Comment (shahadat3669) [inline spec/features/splash_screen_spec.rb:1]: 👍 Great effort on your test cases for the splash screen! To ensure comprehensive coverage, consider extending your integration testing to cover other significant pages, such as the category and payment index pages. Integration tests provide a valuable layer of validation for the overall functionality of your application.

Here's a suggestion for your integration testing strategy:

- [x] **Integration Testing for Category Index Page**: Write test cases to ensure that the category index page functions correctly. This includes checking if categories are displayed as expected, and interactions like adding or deleting categories work seamlessly.

- [x] **Integration Testing for Payment Index Page**: Extend your integration testing to cover the payment index page. Verify that payments are displayed accurately, and any user interactions, such as adding or editing payments, are handled appropriately.

By incorporating integration tests for these critical pages, you'll strengthen the reliability of your application. Keep up the good work, and I'm excited to see the final outcomes! 🚀🧪

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @luigirazum, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ No linter Errors
✅ All the test cases passed


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md
Comment (luigirazum) [reply README.md:1]: Thanks @shahadat3669 for your feedback.
Also I wanted to let you know that I have included the link of the live demo in the readme file and the video presentation to this PR description.

File: spec/features/splash_screen_spec.rb
Comment (luigirazum) [reply spec/features/splash_screen_spec.rb:1]: Thanks again @shahadat3669 for your valuable feedback.
I already added the suggested integration tests, for Categories and Payments page.

**Comment by bellom** [comment]
Hi @luigirazum,

### This is a manual review

# Approved :+1:

Cheers and Happy coding! :clap::clap::clap:

**Pull request merged**